packages/amplify-util-mock/src/CFNParser/appsync-resource-processor.ts

```diff
@@ -0,0 +1,302 @@
+    functions,
+    fieldName: properties.FieldName,
+    requestMappingTemplateLocation: requestMappingTemplate,
+    responseMappingTemplateLocation: responseMappingTemplate,
+    kind: properties.Kind || 'UNIT',
+  };
+}
+
+function getDataSourceName(dataSourceName) {
+  // XXX: Util to map data source based on type of intrinsic function
+  if (typeof dataSourceName === 'string') return dataSourceName;
+  if (isPlainObject(dataSourceName) && Object.keys(dataSourceName).length === 1) {
+    const intrinsicFn = Object.keys(dataSourceName)[0];
+    if (intrinsicFn === 'Fn::GetAtt') {
+      return dataSourceName[intrinsicFn][0];
+    }
+  } else if (dataSourceName.name === 'Fn::ImportValue') {
+    return dataSourceName.payload.payload[1][2];
+  }
+}
+
+function getAppSyncFunctionName(functionConfig) {
+  if (functionConfig['Fn::GetAtt']) {
+    return functionConfig['Fn::GetAtt'][0];
+  }
+  return functionConfig;
+}
+
+export function graphqlFunctionHandler(
+  resourceName,
+  resource,
+  cfnContext: CloudFormationParseContext,
+  transformResult: any
+) {
+  const { Properties: properties } = resource;
+  const requestMappingTemplate = parseValue(
+    properties.RequestMappingTemplateS3Location,
+    cfnContext
+  ).replace('s3://${S3DeploymentBucket}/${S3DeploymentRootKey}/', '');
+  const responseMappingTemplate = parseValue(
+    properties.ResponseMappingTemplateS3Location,
+    cfnContext
+  ).replace('s3://${S3DeploymentBucket}/${S3DeploymentRootKey}/', '');
+
+  const dataSourceName = getDataSourceName(properties.DataSourceName);
+
+  return {
+    name: resource.Properties.Name,
+    dataSourceName,
+    requestMappingTemplateLocation: requestMappingTemplate,
+    responseMappingTemplateLocation: responseMappingTemplate,
+  };
+}
+
+export function processResources(
+  resources,
+  transformResult: any,
+  params = {}
+): AmplifyAppSyncSimulatorConfig {
+  const cfnContext: CloudFormationParseContext = {
+    conditions: {
+      ...CFN_DEFAULT_CONDITIONS,
+    },
+    params: {
+      ...CFN_DEFAULT_PARAMS,
+      env: '${env}',
+      S3DeploymentBucket: '${S3DeploymentBucket}',
+      S3DeploymentRootKey: '${S3DeploymentRootKey}',
+      ...params,
+    },
+    resources: {},
+    exports: {},
+  };
+  const processedResources = {
+    schema: {},
+    resolvers: [],
+    functions: [],
+    dataSources: [],
+    mappingTemplates: [],
+    tables: [],
+    appSync: {
+      name: '',
+      authenticationType: '',
+      apiKey: null,
+    },
+  };
+  Object.entries(resources).forEach(entry => {
+    const [resourceName, resource] = entry;
+    const { Type: resourceType } = resource as any;
+    if (Object.keys(resourceProcessorMapping).includes(resourceType)) {
+      const result = resourceProcessorMapping[resourceType](
+        resourceName,
+        resource,
+        cfnContext,
+        transformResult
+      );
+      cfnContext.resources[resourceName] = result;
+
+      switch (resourceType) {
```

Comment:
What do you think of adding a ```default``` case to print a warning about encountered, but not handled resource types? For example a default AppSync project will have ```Custom::EmptyResource``` or a common one: ```AWS::IAM::Policy```.

Reply:
We don't have any plans to support any custom resources or IAM Policy in local mocking.